src/Service/NomPrenomExtractor.php

```diff
@@ -6,18 +6,24 @@ final class NomPrenomExtractor implements NomPrenomExtractorInterface
 {
     public function extractNomPrenom(string $raw): array
     {
-        $nom = [];
-        $prenom = [];
-        $words = explode(' ', $raw);
+        $raw = $this->uniqueMultipleSpaces($raw);
+        // On extrait le nom et le prénom
+        $return = preg_match("/^(?<nom>[A-ZÀ-Ý]+(?:(?:[\s'\-])*[A-ZÀ-Ý]+)*)\s(?<prenom>[A-ZÀ-Ý][a-zà-ÿ]*(?:(?:[\s'\-])*[A-ZÀ-Ý][a-zà-ÿ]*)*)$/", $raw, $result);
 
-        foreach ($words as $word) {
-            $lastChar = substr($word, -1);
-            mb_strtolower($lastChar, 'UTF-8') === $lastChar ? $prenom[] = $word : $nom[] = $word;
-        }
-
-        return [
-            implode(' ', $nom),
-            implode(' ', $prenom),
+        return 1 !== $return ? ['', ''] :
+        [
+            $result['nom'],
+            $result['prenom'],
         ];
     }
+
+    /**
+     * Permet de supprimer des espaces multiples.
```

Comment:
And also dashes ?

We should rename method I think.

Maybe `removeSpaceAndDashDuplication` ?

Reply:
Can you also move phpdoc on interface ?